### PR TITLE
compression: Optionally use the system's xz cmd

### PR DESCRIPTION
**Pull request by rjoleary**
Using your system's xz command (not the default) will give better
compression when compared to the Go-based compressor. My xz
implementation also appears to be slightly faster.

Signed-off-by: Ryan O'Leary <ryanoleary@google.com>

### CODE REVIEW

**Comment by rjoleary** [comment]
Fixes #217

File: pkg/compression/systemlzma.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	cmd.Stdin = bytes.NewBuffer(decodedData)
+	encodedData, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	// Quoting the XZ Utils manpage:
+	//
+	//     xz supports decompressing .lzma files with or without
+	//     end-of-payload marker, but all.lzma files created by xz will use
```
Comment (GanShun) [inline pkg/compression/systemlzma.go]: all .lzma

File: pkg/compression/x86.go [hunk 2]
```diff
@@ -2,11 +2,23 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-package lzma
+package compression
 
-// DecodeX86 decodes LZMA data with the x86 extension.
-func DecodeX86(encodedData []byte) ([]byte, error) {
-	decodedData, err := Decode(encodedData)
+// LZMAX86 implements Compressor and includes X86 filter which is popular in
```
Comment (GanShun) [inline pkg/compression/x86.go]: includes *the* X86 filter

File: pkg/compression/compression.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+func CompressorFromGUID(guid *guid.GUID) Compressor {
+	switch *guid {
+	case LZMAGUID:
+		if *xzPath != "" {
+			return &SystemLZMA{*xzPath}
+		}
+		return &LZMA{}
+	case LZMAX86GUID:
+		if *xzPath != "" {
+			// Alternatively, the -f86 argument could be passed
```
Comment (GanShun) [inline pkg/compression/compression.go]: The current method is probably a good way to make sure we can still decompress regular xz compressed stuff

**Pull request merged**